ONSdigital/rm-survey-service
# Auto add classifiers

# Motivation and Context

Currently when creating a new survey, classifiers need to be manually added separately. This is variously been achieved with manual database inserts (bad) or using the `/surveys/{surveyId}/classifiers` [POST] endpoint (better, but still manual).

The classifiers for a survey are (currently) always the same, so it makes sense to just generate them when the survey is created.

# What has changed

The `/surveys` [POST] has been updated to create the classifiers as well as creating the core survey.
The `/surveys/{surveyId}/classifiers` [POST] endpoint still exists and functionality is unchanged.

Additionally the `/vendor` folder has been removed and the service upgraded to use `go modules` instead (dependencies now pinned using `go.mod` and `go.sum`). This means a change to the build/release job in concourse to call `go get`

Build step in concourse becomes simply:
```shell
  cd survey-service-pre-release-source
  go get ./...
  make
```

A separate pull request on https://github.com/ONSdigital/ras-deploy will be required.

# How to test?

- Spin up a local environment and create a new survey via the `/create` page of `response-operations-ui`.
- Check the resultant records in the database to see the classifiers have been created and linked to the survey.

# Links

- Fixes #64 to move away from vendor to go modules
- Associated PR: https://github.com/ONSdigital/ras-deploy/pull/110
